packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -0,0 +1,408 @@
+            aria-disabled={ disabled }
+          />
+          <span className={ styles.ArrowBox }>
+            <IconButton
+              className={ css('ms-UpButton', styles.UpButton, (keyboardSpinDirection === KeyboardSpinDirection.up ? styles.active : '')) }
+              disabled={ disabled }
+              iconProps={ incrementButtonIcon }
+              title='Increase'
+              aria-hidden='true'
+              onMouseDown={ () => { this._updateValue(true /* shouldSpin */, this._onIncrement); } }
+              onMouseLeave={ this._stop }
+              onMouseUp={ this._stop }
+              tabIndex={ -1 }
+            />
+            <IconButton
+              className={ css('ms-DownButton', styles.DownButton, (keyboardSpinDirection === KeyboardSpinDirection.down ? styles.active : '')) }
+              disabled={ disabled }
+              iconProps={ decrementButtonIcon }
+              title='Decrease'
+              aria-hidden='true'
+              onMouseDown={ () => { this._updateValue(true /* shouldSpin */, this._onDecrement); } }
+              onMouseLeave={ this._stop }
+              onMouseUp={ this._stop }
+              tabIndex={ -1 }
+            />
+          </span >
+        </div >
+        { labelPosition === Position.bottom && <div className={ styles.labelWrapper } style={ this._labelDirectionHelper() }>
+          { iconProps && <Icon iconName={ iconProps.iconName } className={ css(styles.SpinButtonIcon) } aria-hidden='true'></Icon> }
+          { label &&
+            <Label
+              id={ this._labelId }
+              htmlFor={ this._inputId }
+              className={ styles.SpinButtonLabel }>{ label }
+            </Label>
+          }
+        </div>
+        }
+      </ div >
+    );
+  }
+
+  /**
+   * OnFocus select the contents of the input
+   */
+  public focus() {
+    if (this._spinningByMouse || this.state.keyboardSpinDirection !== KeyboardSpinDirection.notSpinning) {
+      this._stop();
+    }
+
+    this._input.focus();
+    this._input.select();
+  }
+
+  /**
+   * Validate function to use if one is not passed in
+   */
+  private _defaultOnValidate = (value: string) => {
+    if (isNaN(Number(value))) {
+      return this._lastValidValue;
+    }
+    const newValue = Math.min(this.props.max, Math.max(this.props.min, Number(value)));
+    return String(newValue);
+  }
+
+  /**
+   * Increment function to use if one is not passed in
+   */
+  private _defaultOnIncrement = (value: string) => {
+    let newValue = Math.min(Number(value) + this.props.step, this.props.max);
+    return String(newValue);
+  }
+
+  /**
+   * Increment function to use if one is not passed in
+   */
+  private _defaultOnDecrement = (value: string) => {
+    let newValue = Math.max(Number(value) - this.props.step, this.props.min);
+    return String(newValue);
+  }
+
+  /**
+   * Set the label direction with the gap on the correct side
+   */
+  @autobind
+  private _labelDirectionHelper(): any {
```

Comment:
I added a comment before about considering renaming this function to be getLabelDirectionStyle or something like that